Simplified/Accounts.swift

```diff
@@ -64,52 +92,64 @@ func loadDataWithCache(url: URL, cacheUrl: URL, preferringCache: Bool, completio
     }
   }
 
-  fileprivate override init()
-  {
+  fileprivate override init() {
     self.defaults = UserDefaults.standard
+    self.accountSet = NYPLSettings.shared.useBetaLibraries ? betaUrlHash : prodUrlHash
     
     super.init()
     
-    loadCatalogs(preferringCache: true, completion: {_ in })
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(updateAccountSetFromSettings),
+      name: NSNotification.Name.NYPLUseBetaDidChange,
+      object: nil
+    )
+    loadCatalogs(options: .offline, completion: {_ in })
+    loadCatalogs(options: .strict_offline, url: NYPLSettings.shared.useBetaLibraries ? prodUrl : betaUrl, completion: {_ in })
   }
   
   let completionHandlerAccessQueue = DispatchQueue(label: "libraryListCompletionHandlerAccessQueue")
   
   // Returns whether loading was happening already
-  func addLoadingCompletionHandler(_ handler: @escaping (Bool) -> ()) -> Bool {
+  func addLoadingCompletionHandler(key: String, _ handler: @escaping (Bool) -> ()) -> Bool {
     var wasEmpty = false
     completionHandlerAccessQueue.sync {
-      wasEmpty = loadingCompletionHandlers.isEmpty
-      loadingCompletionHandlers.append(handler)
+      if loadingCompletionHandlers[key] == nil {
+        loadingCompletionHandlers[key] = [(Bool)->()]()
+      }
+      wasEmpty = loadingCompletionHandlers[key]!.isEmpty
+      loadingCompletionHandlers[key]!.append(handler)
     }
     return !wasEmpty
   }
   
-  func callAndClearLoadingCompletionHandlers(_ success: Bool) {
+  func callAndClearLoadingCompletionHandlers(key: String, _ success: Bool) {
```

Comment:
Comments on what this functions accomplishes and description of the parameters would be nice :-)